### DISCUSSION TITLE
Data storage / privacy

### DISCUSSION THREAD

**Discussion by sknill-tm** [mesa/mesa#1211] [Q&A]
Hi All, 

Could someone help point me in the direction or advise on whether the mesa package incorporates any data privacy/security measures, especially when running visuals?  I intend to use some sensitive data when running the model and need to understand the intricacies of passing the data through the model before proceeding.  

Appreciate the support!

Cheers

**Comment by tpike3** [comment]
The models are run locally on your machine. The visualization are done via Tornado and a local port. The code and description is [here](https://github.com/projectmesa/mesa/blob/main/mesa/visualization/ModularVisualization.py) .
